### PR TITLE
Orca: fix extra requires in ncf tutorial 

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

1. For `yarn-cluster` mode, extra python lib needed to be provided
2. User always needs `tensorboard` package even when arg `--tensorboard` is not given.

### 2. Summary of the change 

+ train/resume scripts: 
    + remove the global import `TensorBoardCallback`
    + fix `extra_python_lib`
+ `utils.py`: add one function `get_callbacks()`

### CODE REVIEW

File: python/orca/tutorial/NCF/pytorch_resume_train_dataloader.py [hunk 1]
```diff
@@ -24,13 +24,12 @@
...
 
 from bigdl.orca.learn.pytorch import Estimator
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Resume Training with DataLoader")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py,pytorch_dataset.py")
+init_orca(args.cluster_mode, extra_python_lib="pytorch_dataset.py,pytorch_model.py,utils.py")
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_resume_train_dataloader.py:32]: Don't need to pass util.py to extra_python_lib.

File: python/orca/tutorial/NCF/pytorch_resume_train_spark_dataframe.py [hunk 2]
```diff
@@ -23,13 +23,14 @@
...
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Resume Training with Spark DataFrame")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py")
+init_orca(args.cluster_mode, extra_python_lib='process_spark_dataframe.py,'
+                                              'pytorch_model.py,'
+                                              'utils.py')
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_resume_train_spark_dataframe.py:33]: Don't need to pass util.py and process_spark_dataframe.py to extra_python_lib.

File: python/orca/tutorial/NCF/pytorch_resume_train_xshards.py [hunk 3]
```diff
@@ -24,13 +24,12 @@
...
 from bigdl.orca.data import XShards
 from bigdl.orca.learn.pytorch import Estimator
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Resume Training with Orca XShards")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py")
+init_orca(args.cluster_mode, extra_python_lib="process_xshards.py,pytorch_model.py,utils.py")
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_resume_train_xshards.py:32]: Don't need to pass util.py and process_xshard.py to extra_python_lib.

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 4]
```diff
@@ -24,13 +24,12 @@
...
 
 from bigdl.orca.learn.pytorch import Estimator
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Training with DataLoader")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py,pytorch_dataset.py")
+init_orca(args.cluster_mode, extra_python_lib="pytorch_dataset.py,pytorch_model.py,utils.py")
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:32]: the same as above

File: python/orca/tutorial/NCF/pytorch_train_spark_dataframe.py [hunk 5]
```diff
@@ -23,13 +23,14 @@
...
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Training with Spark DataFrame")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py")
+init_orca(args.cluster_mode, extra_python_lib='process_spark_dataframe.py,'
+                                              'pytorch_model.py,'
+                                              'utils.py')
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_train_spark_dataframe.py:33]: the same as above

File: python/orca/tutorial/NCF/pytorch_train_xshards.py [hunk 6]
```diff
@@ -23,13 +23,12 @@
...
 
 from bigdl.orca.learn.pytorch import Estimator
-from bigdl.orca.learn.pytorch.callbacks.tensorboard import TensorBoardCallback
 from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
 
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Training with Orca XShards")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py,process_xshards.py")
+init_orca(args.cluster_mode, extra_python_lib="process_xshards.py,pytorch_model.py,utils.py")
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/pytorch_train_xshards.py:31]: the same as above

File: python/orca/tutorial/NCF/utils.py [hunk 7]
```diff
@@ -111,6 +111,15 @@ def schedule_func(epoch, lr):
         return lr * 0.5
 
 
+def get_callbacks(args):
```
Comment (zpeng1898) [inline python/orca/tutorial/NCF/utils.py:114]: rename it to 'get_pytorch_callbacks'

File: python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py [hunk 8]
```diff
@@ -24,7 +24,9 @@
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Prediction with Spark DataFrame", mode="predict")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py")
+init_orca(args.cluster_mode, extra_python_lib="process_spark_dataframe.py,"
+                                              "pytorch_model.py,"
+                                              "utils.py")
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py:29]: put in one line?

File: python/orca/tutorial/NCF/utils.py [hunk 9]
```diff
@@ -117,6 +117,15 @@ def schedule_func(epoch, lr):
         return lr * 0.5
 
 
+def get_pytorch_callbacks(args):
```
Comment (hkvision) [inline python/orca/tutorial/NCF/utils.py:120]: change args to tensorboard and directly input args.tensorboard?

Comment (hamham223) [reply python/orca/tutorial/NCF/utils.py:120]: Line 123 needs `args.model_dir`
so passing `args` is more simplified?

File: python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py (hunk 8 above)
Comment (hamham223) [reply python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py:29]: It's on the boundary! line too long (101 > 100 characters) 🤣

File: python/orca/tutorial/NCF/utils.py (hunk 9 above)
Comment (hkvision) [reply python/orca/tutorial/NCF/utils.py:120]: okay, make sense

**Comment by hkvision** [comment]
What about tf files?

File: python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py [hunk 10]
```diff
@@ -24,7 +24,8 @@
 
 # Step 1: Init Orca Context
 args = parse_args("PyTorch NCF Prediction with Spark DataFrame", mode="predict")
-init_orca(args.cluster_mode, extra_python_lib="pytorch_model.py")
+init_orca(args.cluster_mode, extra_python_lib="process_spark_dataframe.py,"
+                                              "pytorch_model.py,utils.py")
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_predict_spark_dataframe.py:28]: these are only needed for cluster mode; but we can add them here to be general.

File: python/orca/tutorial/NCF/tf_resume_train_xshards.py [hunk 11]
```diff
@@ -19,16 +19,16 @@
 
 import tensorflow as tf
 
-from utils import *
 from process_xshards import get_feature_cols, get_label_cols
+from utils import *
```
Comment (hamham223) [inline python/orca/tutorial/NCF/tf_resume_train_xshards.py:23]: Keep the order? First import process, then model, then utils

Comment (hkvision) [reply python/orca/tutorial/NCF/tf_resume_train_xshards.py:23]: yep, great

**Pull request merged**